src/sentry/static/sentry/app/components/charts/components/legend.tsx

```diff
@@ -13,26 +13,36 @@ type ChartProps = React.ComponentProps<typeof BaseChart>;
 export default function Legend(
   props: ChartProps['legend'] & {theme?: Theme} = {}
 ): EChartOption.Legend {
-  const {truncate, theme, ...rest} = props ?? {};
+  const {truncate, theme, textStyle, ...rest} = props ?? {};
   const formatter = (value: string) => truncationFormatter(value, truncate ?? 0);
 
-  return {
+  const newTextStyle = {
+    color: theme?.textColor,
+    verticalAlign: 'top',
+    fontSize: 11,
+    fontFamily: 'Rubik',
+    ...textStyle,
+  };
+
+  const legend = {
     show: true,
-    type: 'scroll',
+    type: 'scroll' as const,
     padding: 0,
     formatter,
     icon: 'circle',
     itemHeight: 8,
     itemWidth: 8,
     itemGap: 12,
     align: 'left' as const,
-    textStyle: {
-      color: theme?.textColor,
-      verticalAlign: 'top',
-      fontSize: 11,
-      fontFamily: 'Rubik',
-    },
+    textStyle: newTextStyle,
     inactiveColor: theme?.inactive,
     ...rest,
   };
+
+  if (theme === undefined) {
+    delete newTextStyle.color;
+    delete legend.inactiveColor;
+  }
```

Comment:
Would this make the echarts use the default value which is a black? Or do we have a color defined in global css?

Reply:
Because `Legend` is always called in `BaseChart` with the appropriate theme, the colours will be set correctly. Thinking about it now, this is very implicit. I'll see if there is some way of making this more explicit.

Reply:
Seems like we shouldn't be calling `Legend()` in chart components. I added more calls last week, but those should be reverted if BaseChart is always using `Legend()`.